modules/01_Theory.md

```diff
@@ -0,0 +1,409 @@
+
+```ts
+import { add } from "./math.mjs";
+add(1, 2);
+```
+
+might be emitted as either:
+
+```ts
+import { add } from "./math.mjs";
+add(1, 2);
+```
+
+or:
+
+```ts
+const math_1 = require("./math.mjs");
+math_1.add(1, 2);
+```
+
+depending on the `module` compiler option, but the module specifier will always be `"./math.mjs"`. There is no compiler option that enables transforming, substituting, or rewriting module specifiers. Consequently, module specifiers must be written in a way that works for the code’s target runtime or bundler, and it’s TypeScript’s job to understand those _output_-relative specifiers. The process of finding the file referenced by a module specifier is called _module resolution_.
+
+## Module resolution
+
+Let’s return to our [first example](#typescripts-job-concerning-modules) and review what we’ve learned about it so far:
+
+```ts
+import sayHello from "greetings";
+sayHello("world");
+```
+
+So far, we’ve discussed how the host’s module system and TypeScript’s `module` compiler option might impact this code: we know that the input syntax looks like ESM, but the output format depends on the `module` compiler option and potentially the file extension and `package.json` `"type"` field; we also know that what `sayHello` gets bound to, and even whether the import is even allowed, may vary depending on the module kinds of this file and the target file. But we haven’t yet discussed how to _find_ the target file.
+
+### Module resolution is host-defined
+
+While the ECMAScript specification defines how to parse and interpret `import` and `export` statements, it leaves module resolution up to the host. If you’re creating a hot new JavaScript runtime, you’re free to create a module resolution scheme like:
+
+```ts
+import monkey from "🐒"; // Looks for './eats/bananas.js'
+import cow from "🐄";    // Looks for './eats/grass.js'
+import lion from "🦁";   // Looks for './eats/you.js'
+```
+
+and still claim to implement “standards-compliant ESM.” Needless to say, TypeScript would have no idea what types to assign to `monkey`, `cow`, and `lion` without built-in knowledge of this runtime’s module resolution algorithm. Just as `module` informs the compiler about the host’s expected module format, `moduleResolution`, along with a few customization options, specify the algorithm the host uses to resolve module specifiers to files.
+
+The available `moduleResolution` options are:
+
+- [**`classic`**](./03_Reference.md#classic): TypeScript’s oldest and most inscrutable module resolution mode, this is unfortunately the default when `module` is set to anything other than `commonjs`, `node16`, or `nodenext`. It does not represent any real-world module resolver and should never be used. It was kept around only for backwards compatibility.
```

Comment:
```suggestion
- [**`classic`**](./03_Reference.md#classic): TypeScript’s oldest and most inscrutable module resolution mode, this is unfortunately the default when `module` is set to anything other than `commonjs`, `node16`, or `nodenext`. It does not represent any real-world module resolver and should never be used. It was kept around only for backwards compatibility. This option is scheduled for deprecation in TypeScript 6.0.
```

Reply:
I took a marginally higher effort stab at figuring out why this exists in some new edits on the website PR, but will add the deprecation note as well:

> TypeScript’s oldest module resolution mode, this is unfortunately the default when `module` is set to anything other than `commonjs`, `node16`, or `nodenext`. It was probably made to provide best-effort resolution for a wide range of [RequireJS](https://requirejs.org/docs/api.html#packages) configurations. It should not be used for new projects (or even old projects that don’t use RequireJS or another AMD module loader).